calblueprint/siren-web
# Andrei/intake home

# ✨ New in this PR ✨

## One liner: what did you do? 🚨 
Home page for intake forms/ case types
![image](https://user-images.githubusercontent.com/73627326/172101316-312340c8-a5ac-4afe-80b9-2ca7dafcb781.png)


## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [x] Load all case types + create new forms for firebase
 - [x] Rename existing cases
 - [x] delete button to remove case from fb
 - [ ] styling + documentation for intake forms

## How can the reviewer test your code? 👩‍💻
_On intake form tab, load in cases or create new cases.  Rename existing cases in the intake home tab, make sure that their questions are preserved afterwards.  In case of accidental deletion, import from caseTypesFullBackup from this [link](https://console.cloud.google.com/firestore/import-export?project=siren-6099f)_

## Any bugs you encountered or still having trouble with? 🐛
_When renaming, a new case is created and the previously named case has all of its contents moved into the new path before deleting the old path.  This takes a moment so questions will be blank until the copying is finished, when promises are fulfilled there is a forced refresh.  Would look smoother if useEffect remounting would work, but unable to find a variable that would work best as a dependency.


## Resources 📔
_link online resources and related PRs_ 


🧜‍♀️ cc: @shannonbonet @phoebeli23
